lib/trollop.rb

```diff
@@ -124,7 +371,7 @@ def opt(name, desc = "", opts = {}, &b)
     opts[:callback] ||= b if block_given?
     opts[:desc] ||= desc
 
-    o = Option.create(name, desc, opts)
+    o = OptionDispatch.create(name, desc, opts, @settings)
```

Comment:
Do we need a separate `OptionDispatch` class vs following Rails conventions of the factory being in the root class?

Guess I'm not totally sure what this class does

Reply:
I'm unfamiliar w/ Rails conventions, but think i follow your idea.  This can probably be a class method on `Option` and not need a separate factory class.